CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -752,38 +756,37 @@ function Humanoid:_handleEmptyActionQueue()
     self:queueAction(MeanderAction())
   elseif class.is(self, GrimReaper) then
     self:queueAction(IdleAction())
-  else
+  elseif class.is(self, Patient) then
     self:queueAction(SeekReceptionAction())
+  else
+    self:queueAction(IdleAction())
   end
 
-  -- Open the dialog of the humanoid as feedback to the user.
-  local ui = self.world.ui
-  if class.is(self, Patient) then
-    ui:addWindow(UIPatient(ui, self))
-  elseif class.is(self, Staff) then
-    ui:addWindow(UIStaff(ui, self))
-  end
+  if TheApp.config.debug then
+    -- Open the dialog of the humanoid as feedback to the user.
+    local ui = self.world.ui
+    if class.is(self, Patient) then
+      ui:addWindow(UIPatient(ui, self))
+    elseif class.is(self, Staff) then
+      ui:addWindow(UIStaff(ui, self))
+    end
 
-  -- Tell the player what just happened.
-  self.world:gameLog("")
-  self.world:gameLog("Empty action queue!")
-  self.world:gameLog("Last action: " .. self.previous_action.name)
-  self.world:gameLog(debug.traceback())
-
-  ui:addWindow(UIConfirmDialog(ui, true, _S.errors.dialog_empty_queue,
-    --[[persistable:humanoid_leave_hospital]] function()
-      self.world:gameLog("The humanoid was told to leave the hospital...")
-      if class.is(self, Staff) then
-        self:fire()
-      else
-        -- Set these variables to increase the likelihood of the humanoid managing to get out of the hospital.
-        self.going_home = false
-        self.hospital = self.world:getLocalPlayerHospital()
-        self:goHome("kicked")
-      end
-    end,
-    nil -- Do nothing on cancel
-  ))
+    -- Ask user to choose kick off humanoid or do nothing
+    ui:addWindow(UIConfirmDialog(ui, true, _S.errors.dialog_empty_queue,
+      --[[persistable:humanoid_leave_hospital]] function()
+        self.world:gameLog("The humanoid was told to leave the hospital...")
+        if class.is(self, Staff) then
+          self:fire()
+        else
+          -- Set these variables to increase the likelihood of the humanoid managing to get out of the hospital.
+          self.going_home = false
+          self.hospital = self.world:getLocalPlayerHospital()
+          self:goHome("kicked")
+        end
+      end,
+      nil -- Do nothing on cancel
+    ))
+  end
```

Comment:
else:
print("Warning: Humanoid encountered an empty action queue. Recovery has been attempted.")

Reply:
Print() will output to the console, if one is present
Gamelog will log into the table of things to write on dump.

However, if you call gamelog with debug on, every thing ever added to the table is also printed to the console.

Code for gamelog addition:
```lua
--! Append a message to the game log.
--!param message (string) The message to add.
function World:gameLog(message)
  self.game_log[#self.game_log + 1] = message
  -- If in debug mode also show it in the command prompt
  if TheApp.config.debug then
    print(message)
  end
end
```

Reply:
The print is only really useful for Windows and some Linux users.... Mac probably has no care for the terminal unless you go out your way to launch the game with one.